src/database/mongo/sorted.js

```diff
@@ -533,4 +533,40 @@ module.exports = function(db, module) {
 				callback(err, data);
 		});
 	};
-};
\ No newline at end of file
+
+	module.processSortedSet = function(setKey, process, batch, callback) {
+		var done = false, ids = [], cursor = db.collection('objects').
+			find({_key: setKey}).
+			sort({score: 1}).
+			project({_id: 0, value: 1}).
+			batchSize(batch);
+
+		async.whilst(
+			function() {
+				return !done;
+			},
+			function(next) {
+				cursor.next(function(err, item) {
+					if (err) {
+						return next(err);
+					}
+					if (item === null) {
+						done = true;
```

Comment:
The first iteration gets the element and puts it into the ids array. The second iteration gets null for item, so it sets done and calls process, then calls next from the callback for process.